engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -99,8 +100,8 @@ public void processRecord(
         processState.getProcessByKey(processInstanceRecord.getProcessDefinitionKey());
 
     final var optRejection = validateCommand(command, process);
-    if (optRejection.isPresent()) {
-      final var rejection = optRejection.get();
+    if (optRejection.isLeft()) {
+      final var rejection = optRejection.getLeft();
```

Comment:
🔧 Changing Optional to Either probably has a good reason, but it's not documented. This is a good case where the reasoning could be described in the commit message.